### PR TITLE
Migration

### PR DESCRIPTION
**Pull request by Eternalgratis**
- [ ] Create a new app. Give it a meaningful name.
- [ ] Initialize your project with Git.
- [ ] Make sure that your project has Postgres database set up.
- [ ] Build your project schema
- Create and run the necessary migration files.
- Table and column names should match the ERD diagram.
- photo for users table should be a link to an image
- Foreign keys should be included.

All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  create_table 'posts', force: :cascade do |t|
+    t.string 'title'
+    t.text 'text'
+    t.integer 'comments_counter'
+    t.integer 'likes_counter'
+    t.datetime 'created_at', null: false
+    t.datetime 'updated_at', null: false
+    t.bigint 'user_id', null: false
```
Comment (SajjadAhmad14) [inline db/schema.rb:45]: Please add `author_id` as a foreign key instead of `user_id` as it's a requirement in the project.

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    t.bigint 'post_id', null: false
+    t.index ['post_id'], name: 'index_comments_on_post_id'
+    t.index ['user_id'], name: 'index_comments_on_user_id'
+  end
+
+  create_table 'likes', force: :cascade do |t|
+    t.datetime 'created_at', null: false
+    t.datetime 'updated_at', null: false
+    t.bigint 'user_id', null: false
```
Comment (SajjadAhmad14) [inline db/schema.rb:32]: Please add `author_id` as a foreign key instead of `user_id` as it's a requirement in the project.

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+ActiveRecord::Schema[7.0].define(version: 20_221_025_211_212) do
+  # These are extensions that must be enabled in order to support this database
+  enable_extension 'plpgsql'
+
+  create_table 'comments', force: :cascade do |t|
+    t.text 'text'
+    t.datetime 'created_at', null: false
+    t.datetime 'updated_at', null: false
+    t.bigint 'user_id', null: false
```
Comment (SajjadAhmad14) [inline db/schema.rb:23]: Please add `author_id` as a foreign key instead of `user_id` as it's a requirement in the project.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Titi, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
PR has a title and summary. :heavy_check_mark:
## To Improve 🙏

- [ ] Please add a professional README file to your project. A professional README should have a project name, project title, and a set of instructions to run the project on a local machine.

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/schema.rb (hunk 1 above)
Comment (Eternalgratis) [reply db/schema.rb:45]: @SajjadAhmad14 thanks for the feedback but according to the ERD diagram the other tables are referencing from the user table
We have the 
1. User table, post table, comment, and like 
where post, comment and like are referencing from the user table
- If i am to change the user_id to author_id then my User table will have to be changed to Author and i feel that does not go inline with the ERD diagram 

Kindly see to this and give feedback on what you suggest its best i do in this instance

See below the diagram
ERM

Comment (Eternalgratis) [reply db/schema.rb:45]: @SajjadAhmad14 thanks for the feedback but according to the ERD diagram the other tables are referencing from the user table
We have the 
1. User table, post table, comment, and like 
where post, comment and like are referencing from the user table
- If i am to change the user_id to author_id then my User table will have to be changed to Author and i feel that does not go inline with the ERD diagram 

Kindly see to this and give feedback on what you suggest its best i do in this instance

File: db/schema.rb (hunk 2 above)
Comment (Eternalgratis) [reply db/schema.rb:32]: @SajjadAhmad14 thanks for the feedback but according to the ERD diagram the other tables are referencing from the user table
We have the 
1. User table, post table, comment, and like 
where post, comment and like are referencing from the user table
- If i am to change the user_id to author_id then my User table will have to be changed to Author and i feel that does not go inline with the ERD diagram 

Kindly see to this and give feedback on what you suggest its best i do in this instance

See below the diagram
ERM

File: db/schema.rb (hunk 3 above)
Comment (Eternalgratis) [reply db/schema.rb:23]: @SajjadAhmad14 thanks for the feedback but according to the ERD diagram the other tables are referencing from the user table
We have the 
1. User table, post table, comment, and like 
where post, comment and like are referencing from the user table
- If i am to change the user_id to author_id then my User table will have to be changed to Author and i feel that does not go inline with the ERD diagram 

Kindly see to this and give feedback on what you suggest its best i do in this instance

See below the diagram
ERM

**Review by m05e5 — APPROVED** [review]
Hi @Eternalgratis , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ Descriptive PR 

## Optional suggestions

- [ ] [ optional ] **you're right about the naming convention, but when working on the backend 
 of a project, you have to follow exactly what's on your ER Diagram because in the future you may be working with a team and that's what everyone will take as a reference**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**